jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -1076,10 +1076,19 @@ void java_object_factoryt::gen_nondet_init(
     // add assumes to obey numerical restrictions
     if(type != java_boolean_type() && type != java_char_type())
     {
-      exprt within_bounds = object_factory_parameters.assume_inputs_interval
-        .make_contains_expr(expr);
-      if(within_bounds != true_exprt{})
-        assignments.add(code_assumet(std::move(within_bounds)));
+      const auto &interval = object_factory_parameters.assume_inputs_interval;
+      if(auto singleton = interval.as_singleton())
+      {
+        assignments.add(
+          code_assignt{expr, from_integer(*singleton, expr.type())});
```

Comment:
:grey_question: Can this actually override the original value? Does that matter?

Reply:
the original value is nondet, so it doesn't matter